cmake/SDFUtils.cmake

```
@@ -52,8 +52,10 @@ endmacro (BUILD_WARNING)
 #################################################
 macro (sdf_add_library _name)
   set(LIBS_DESTINATION ${PROJECT_BINARY_DIR}/src)
-  set_source_files_properties(${ARGN} PROPERTIES COMPILE_DEFINITIONS "BUILDING_DLL")
-  add_library(${_name} SHARED ${ARGN})
+  if(BUILD_SHARED_LIBS)
+    set_source_files_properties(${ARGN} PROPERTIES COMPILE_DEFINITIONS "BUILDING_DLL")
+  endif()
```

Using CMake's [`DEFINE_SYMBOL`](https://cmake.org/cmake/help/latest/prop_tgt/DEFINE_SYMBOL.html), you can do:
```suggestion
  set_target_properties( ${_name} PROPERTIES DEFINE_SYMBOL "BUILDING_DLL")
```